### PR TITLE
Fix a bug that throws a NotSupportedException when Integrated Security keyword is used at all in SqlConnections on non-Windows platforms. It should only fail when set to true.

### PR DESCRIPTION
**Pull request by corivera**
Fixes: https://github.com/aspnet/EntityFramework/issues/4915
Tracking bug: https://github.com/dotnet/corefx/issues/7417

### CODE REVIEW

**Comment by corivera** [comment]
CC @saurabh500 @kkurni

File: src/System.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs [hunk 1]
```diff
@@ -19,6 +19,19 @@ public static class ExceptionTest
...
+#if MANAGED_SNI
+        [Fact]
+        public static void NonWindowsIntAuthFailureTest()
+        {
+            string connectionString = DataTestClass.SQL2008_Northwind + ";Integrated Security = true";
+            Assert.Throws<NotSupportedException>(() => new SqlConnection(connectionString).Open());
+
+            // Should not receive any exception when using IntAuth=false
+            connectionString = DataTestClass.SQL2008_Northwind + ";Integrated Security = false";
+            new SqlConnection(connectionString).Open();
```
Comment (saurabh500) [inline src/System.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs]: Everything looks good overall. How about a test without IntAuth at all?

**Comment by mmitche** [comment]
@dotnet-bot test this please

File: src/System.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs (hunk 1 above)
Comment (corivera) [inline src/System.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs]: That should be covered by the rest of our tests.

**Pull request merged**